packages/squiggle-lang/src/rescript/Distributions/GenericDist/GenericDist.res

```diff
@@ -300,27 +301,38 @@ module AlgebraicCombination = {
     ~arithmeticOperation: Operation.algebraicOperation,
     ~t2: t,
   ): result<t, error> => {
-    switch strategy {
-    | AsDefault => runDefault(t1, ~toPointSetFn, ~toSampleSetFn, ~arithmeticOperation, ~t2)
-    | AsSymbolic =>
-      switch tryAnalyticalSimplification(arithmeticOperation, t1, t2) {
-      | Some(#AnalyticalSolution(symbolicDist)) => Ok(Symbolic(symbolicDist))
-      | Some(#NoSolution) => Error(RequestedStrategyInvalidError(`No analytical solution`))
-      | None => Error(RequestedStrategyInvalidError("Inputs were not even symbolic"))
-      | Some(#Error(err)) => Error(OperationError(err))
+    let invalidOperationError = InputValidator.run(t1, t2, ~arithmeticOperation, ~toPointSetFn)
+    switch (invalidOperationError, strategy) {
+    | (Some(e), _) => Error(e)
+    | (None, AsDefault) => {
```

Comment:
Any reason these constructors (`AsDefault`, etc.) wouldn't be `StrategyChooser.specificStrategy`? Can't you say it's of type `[< #AsDefault | StrategyChooser.specificStrategy]` or something?

Reply:
I agree this could be improved. Will consider for this PR.